README.md

```diff
@@ -1 +1,523 @@
-# IPVM Workflow Specification
+# IPVM Workflow Specification v0.1.0
+
+## Editors
+
+* [Brooklyn Zelenka](https://github.com/expede), [Fission](https://fission.codes)
+
+## Authors
+
+* [Brooklyn Zelenka](https://github.com/expede), [Fission](https://fission.codes)
+* [Simon Worthington](https://github.com/simonwo), [Bacalhau Project](https://www.bacalhau.org/)
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Dependencies
+
+* [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/spec/)
+* [UCAN Invocation](https://github.com/ucan-wg/invocation/)
+* [VarSig](https://github.com/ChainAgnostic/varsig/)
+
+# 0 Abstract
+
+An IPVM Workflow is a declarative configuration that extends a [UCAN Invocation](https://github.com/ucan-wg/invocation). A Workflow provides everything required to execute one or more tasks: defaults, tasks and their dependencies, authorization, metadata, signatures, and so on.
+
+# 1 Introduction
+
+> In late 1970 or early ’71 I approached IBM Canada’s Intellectual Property department to see if we could take out a patent on the basic idea [of dataflow]. Their recommendation, which I feel was prescient, was that this concept seemed to them more like a law of nature, which is not patentable. 
+>
+> J. Paul Morrison, [Flow-Based Programming](https://jpaulm.github.io/fbp/book.html)
+
+The potential complexity of a fully distributed execution by untrusted peers is very high. IPVM Workflows reduce the number of possible states by forcing explicit handling of any dangerous effects. The IPVM Workflow spec is a declarative document that MAY be inspected, transmitted, logged, and negotiated. Unlike s systems like WASI, there is a strict separation of effects from pure data, an emphasis on verifiability, and [promise pipelining](http://erights.org/elib/distrib/pipeline.html).
```

Comment:
Q: are effects still getting their own write-up?

Reply:
(For those following along at home: Zeeshan and I chatted about this live)

They are, absolutely! I'm trying to keep the various specs as separate as possible; my current hypothesis is that we're going to do the scheduler safety properties for URI/Ability pairs as part of service discovery. They'll need to register something like:

```haskell
Service {
  scheme: "https"
  ability: "crud/read"
  idempotent: false,
  destructive: false,
  max_verifiability: Aattestation
}
```

We know that everything is some kind of effect _unless_ it's raw Wasm, so that makes it really easy for the runtime to detect. If you want to run e.g. a pure subset of Python, you can definitely get an interpreter written in Wasm to feed the source into. We could open up `pure: Bool` as well, but the tooling for Wasm is advancing so quickly that I _currently_ don't believe that we'll need anything beyond "run the source inside Wasm"